SocialCareCaseViewerApi/V1/Controllers/SocialCareCaseViewerApiController.cs

```diff
@@ -127,6 +127,25 @@ public IActionResult ListCases([FromQuery] ListCasesRequest request)
             }
         }
 
+        /// <summary>
+        /// Find specific case by unique Record ID produced by MongoDB
+        /// </summary>
+        /// <response code="200">Success. Returns case related to the specified ID</response>
+        /// <response code="404">No cases found for the specified ID or officer email</response>
+        [ProducesResponseType(typeof(CareCaseDataList), StatusCodes.Status200OK)]
+        [HttpGet]
+        [Route("cases/{recordId}")]
+        public IActionResult GetCaseByRecordId(string recordId)
+        {
+            try
+            {
+                return Ok(_processDataUsecase.Execute(recordId));
+            }
+            catch (DocumentNotFoundException e)
```

Comment:
If this exception based pattern is used (like elsewhere), please ensure the exception is thrown when document is not found so it can be caught here.

Reply:
Exception null response added